lib/formatter.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const generateFunction = require('generate-function');
+
+function getFormatter(metric, defaultLabels) {
+	const defaultLabelNames = Object.keys(defaultLabels);
+	const name = metric.name;
+	const help = escapeString(metric.help);
+	const type = metric.type;
+	const labelsCode = getLabelsCode(metric, defaultLabelNames, defaultLabels);
```

Comment:
⭐ This is going to make it impossible to implement #298/landing #368. As you can tell from the comments in those, I'm not certain what the fate of that issue is though.

Reply:
Can you clarify what do you mean? Should I rewrite it or implement inside generated function?

Reply:
@zbjornson can you take a look on this?

Reply:
Thanks for the nudge, will review this PR this weekend.

On this particular comment: I'd (strongly) lean toward making the formatter work with dynamically declared labels. #298 has quite a bit of support so I suspect #368 will land. The main (probably minor) thing holding it up is https://github.com/siimon/prom-client/issues/298#issuecomment-620477855.

Reply:
This eliminate the main idea of `formatter` - build labels string once.
I need to think how to rewrite this.